xbmc/cores/RetroPlayer/buffers/BaseRenderBufferPool.cpp

```diff
@@ -82,29 +82,21 @@ IRenderBuffer *CBaseRenderBufferPool::GetBuffer(unsigned int width, unsigned int
   {
     CSingleLock lock(m_bufferMutex);
 
-    while (!m_free.empty())
+    for (auto it = m_free.begin(); it != m_free.end(); ++it)
     {
-      renderBuffer = m_free.front().release();
-      m_free.pop_front();
+      std::unique_ptr<IRenderBuffer> &buffer = m_free.front();
 
       // Only return buffers of the same dimensions
-      const unsigned int bufferWidth = renderBuffer->GetWidth();
-      const unsigned int bufferHeight = renderBuffer->GetHeight();
+      const unsigned int bufferWidth = buffer->GetWidth();
+      const unsigned int bufferHeight = buffer->GetHeight();
 
       if (bufferWidth == width && bufferHeight == height)
       {
+        renderBuffer = buffer.release();
         renderBuffer->SetHeader(header);
+        m_free.erase(it);
```

Comment:
it = m_free.erase(it); i believe ?

Reply:
then you jump over one element because erase returns next element and for loop (`++it`) also increments

Reply:
yes, but it's undefined what ++it points to in that case.

Reply:
yes, but it is undefined what ++it points to in that case.

Reply:
Actually you don't have to care about `it`, see next line, loop is left anyway.

Reply:
right